fedimint-server/src/net/api.rs

```diff
@@ -551,6 +557,15 @@ pub fn server_endpoints() -> Vec<ApiEndpoint<ConsensusApi>> {
                 Ok((&fedimint.session_status(index).await).into())
             }
         },
+        api_endpoint! {
+            SHUTDOWN_ENDPOINT,
+            ApiVersion::new(0, 0),
+            async |fedimint: &ConsensusApi, context, index: Option<u64>| -> () {
+                check_auth(context)?;
+                fedimint.shutdown(index);
```

Comment:
Ideally here we should check against current session count and error out if it's too close or past it. This is to prevent races. We should require that the shutdown is at least 1 or maybe 2 sessions in the future, so that delay between here and consensus thread can't make it miss the shutdown. Also persist.